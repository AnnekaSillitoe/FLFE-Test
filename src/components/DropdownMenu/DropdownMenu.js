import React, {Component} from "react";

class DropdownMenu extends Component {
  state = {
    active: ""
  };

  setActive = (e) => {
    e.preventDefault();
    this.setState({
      active: e.target.name
    });
  };

  render() {
    return (
      <div className="dropdown-box">
        <div className="dropdown-triangle"/>
        <div className="dropdown-menu">
          <a href="/" className={`dropdown-header ${this.state.active === "home" ? "active" : ""}`} name="home" onClick={(e) => this.setActive(e)}>
            <div className="icon icon-home"/>
            FutureLearn
          </a>
          <div className="dropdown-main-menu">
            <a href="/" className={`dropdown-item ${this.state.active === "create" ? "active" : ""}`} name="create" onClick={(e) => this.setActive(e)}>
              <div className="icon icon-create"/>
              Course Creator</a>
            <a href="/" className={`dropdown-item ${this.state.active === "admin" ? "active" : ""}`} name="admin" onClick={(e) => this.setActive(e)}>
              <div className="icon icon-admin"/>
              Admin</a>
          </div>
        </div>
      </div>
    );
  }
}

export default DropdownMenu;
