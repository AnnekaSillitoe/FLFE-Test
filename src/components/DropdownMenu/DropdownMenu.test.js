import React from "react";
import { shallow } from "enzyme";
import DropdownMenu from "./DropdownMenu";

test("DropdownMenu renders correctly", () => {
  const dropdownMenu = shallow(<DropdownMenu />);
  expect(dropdownMenu).toMatchSnapshot();
});

test("renders the correct text for each dropdown menu idea", () => {
  const dropdownMenu = shallow(<DropdownMenu />);
  const a1 = dropdownMenu.find("a").first();
  const name1 = a1.text();
  const a2 = dropdownMenu.find("a").at(1);
  const name2 = a2.text();
  const a3 = dropdownMenu.find("a").at(2);
  const name3 = a3.text();

  expect(name1).toEqual("FutureLearn");
  expect(name2).toEqual("Course Creator");
  expect(name3).toEqual("Admin");
});
