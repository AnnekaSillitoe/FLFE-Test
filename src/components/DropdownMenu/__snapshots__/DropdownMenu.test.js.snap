// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropdownMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="dropdown-triangle"
        />,
        <div
          className="dropdown-menu"
        >
          <a
            className="dropdown-header "
            href="/"
            name="home"
            onClick={[Function]}
          >
            <div
              className="icon icon-home"
            />
            FutureLearn
          </a>
          <div
            className="dropdown-main-menu"
          >
            <a
              className="dropdown-item "
              href="/"
              name="create"
              onClick={[Function]}
            >
              <div
                className="icon icon-create"
              />
              Course Creator
            </a>
            <a
              className="dropdown-item "
              href="/"
              name="admin"
              onClick={[Function]}
            >
              <div
                className="icon icon-admin"
              />
              Admin
            </a>
          </div>
        </div>,
      ],
      "className": "dropdown-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "dropdown-triangle",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="dropdown-header "
              href="/"
              name="home"
              onClick={[Function]}
            >
              <div
                className="icon icon-home"
              />
              FutureLearn
            </a>,
            <div
              className="dropdown-main-menu"
            >
              <a
                className="dropdown-item "
                href="/"
                name="create"
                onClick={[Function]}
              >
                <div
                  className="icon icon-create"
                />
                Course Creator
              </a>
              <a
                className="dropdown-item "
                href="/"
                name="admin"
                onClick={[Function]}
              >
                <div
                  className="icon icon-admin"
                />
                Admin
              </a>
            </div>,
          ],
          "className": "dropdown-menu",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="icon icon-home"
                />,
                "FutureLearn",
              ],
              "className": "dropdown-header ",
              "href": "/",
              "name": "home",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon icon-home",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "FutureLearn",
            ],
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className="dropdown-item "
                  href="/"
                  name="create"
                  onClick={[Function]}
                >
                  <div
                    className="icon icon-create"
                  />
                  Course Creator
                </a>,
                <a
                  className="dropdown-item "
                  href="/"
                  name="admin"
                  onClick={[Function]}
                >
                  <div
                    className="icon icon-admin"
                  />
                  Admin
                </a>,
              ],
              "className": "dropdown-main-menu",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="icon icon-create"
                    />,
                    "Course Creator",
                  ],
                  "className": "dropdown-item ",
                  "href": "/",
                  "name": "create",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "icon icon-create",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  "Course Creator",
                ],
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="icon icon-admin"
                    />,
                    "Admin",
                  ],
                  "className": "dropdown-item ",
                  "href": "/",
                  "name": "admin",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "icon icon-admin",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  "Admin",
                ],
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="dropdown-triangle"
          />,
          <div
            className="dropdown-menu"
          >
            <a
              className="dropdown-header "
              href="/"
              name="home"
              onClick={[Function]}
            >
              <div
                className="icon icon-home"
              />
              FutureLearn
            </a>
            <div
              className="dropdown-main-menu"
            >
              <a
                className="dropdown-item "
                href="/"
                name="create"
                onClick={[Function]}
              >
                <div
                  className="icon icon-create"
                />
                Course Creator
              </a>
              <a
                className="dropdown-item "
                href="/"
                name="admin"
                onClick={[Function]}
              >
                <div
                  className="icon icon-admin"
                />
                Admin
              </a>
            </div>
          </div>,
        ],
        "className": "dropdown-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "dropdown-triangle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="dropdown-header "
                href="/"
                name="home"
                onClick={[Function]}
              >
                <div
                  className="icon icon-home"
                />
                FutureLearn
              </a>,
              <div
                className="dropdown-main-menu"
              >
                <a
                  className="dropdown-item "
                  href="/"
                  name="create"
                  onClick={[Function]}
                >
                  <div
                    className="icon icon-create"
                  />
                  Course Creator
                </a>
                <a
                  className="dropdown-item "
                  href="/"
                  name="admin"
                  onClick={[Function]}
                >
                  <div
                    className="icon icon-admin"
                  />
                  Admin
                </a>
              </div>,
            ],
            "className": "dropdown-menu",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="icon icon-home"
                  />,
                  "FutureLearn",
                ],
                "className": "dropdown-header ",
                "href": "/",
                "name": "home",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon icon-home",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "FutureLearn",
              ],
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    className="dropdown-item "
                    href="/"
                    name="create"
                    onClick={[Function]}
                  >
                    <div
                      className="icon icon-create"
                    />
                    Course Creator
                  </a>,
                  <a
                    className="dropdown-item "
                    href="/"
                    name="admin"
                    onClick={[Function]}
                  >
                    <div
                      className="icon icon-admin"
                    />
                    Admin
                  </a>,
                ],
                "className": "dropdown-main-menu",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="icon icon-create"
                      />,
                      "Course Creator",
                    ],
                    "className": "dropdown-item ",
                    "href": "/",
                    "name": "create",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "icon icon-create",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                    "Course Creator",
                  ],
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="icon icon-admin"
                      />,
                      "Admin",
                    ],
                    "className": "dropdown-item ",
                    "href": "/",
                    "name": "admin",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "icon icon-admin",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                    "Admin",
                  ],
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
